src/native/fs/fs_napi.cpp

```diff
@@ -213,6 +214,112 @@ struct Stat : public FSWorker
+        res["ino"] = Napi::Number::New(env, _stat_res.st_ino);
+        res["mode"] = Napi::Number::New(env, _stat_res.st_mode);
+        res["nlink"] = Napi::Number::New(env, _stat_res.st_nlink);
+        res["uid"] = Napi::Number::New(env, _stat_res.st_uid);
+        res["gid"] = Napi::Number::New(env, _stat_res.st_gid);
+        res["rdev"] = Napi::Number::New(env, _stat_res.st_rdev);
+        res["size"] = Napi::Number::New(env, _stat_res.st_size);
+        res["blksize"] = Napi::Number::New(env, _stat_res.st_blksize);
+        res["blocks"] = Napi::Number::New(env, _stat_res.st_blocks);
+
+        // https://nodejs.org/dist/latest-v14.x/docs/api/fs.html#fs_stat_time_values
+#ifdef __APPLE__
+        double atimeMs = (double(1e3) * _stat_res.st_atimespec.tv_sec) + (double(1e-6) * _stat_res.st_atimespec.tv_nsec);
+        double ctimeMs = (double(1e3) * _stat_res.st_ctimespec.tv_sec) + (double(1e-6) * _stat_res.st_ctimespec.tv_nsec);
+        double mtimeMs = (double(1e3) * _stat_res.st_mtimespec.tv_sec) + (double(1e-6) * _stat_res.st_mtimespec.tv_nsec);
+        double birthtimeMs = (double(1e3) * _stat_res.st_birthtimespec.tv_sec) + (double(1e-6) * _stat_res.st_birthtimespec.tv_nsec);
+#else
+        double atimeMs = (double(1e3) * _stat_res.st_atim.tv_sec) + (double(1e-6) * _stat_res.st_atim.tv_nsec);
+        double ctimeMs = (double(1e3) * _stat_res.st_ctim.tv_sec) + (double(1e-6) * _stat_res.st_ctim.tv_nsec);
+        double mtimeMs = (double(1e3) * _stat_res.st_mtim.tv_sec) + (double(1e-6) * _stat_res.st_mtim.tv_nsec);
+        double birthtimeMs = ctimeMs; // Posix doesn't have birthtime
+#endif
+
+        res["atimeMs"] = Napi::Number::New(env, atimeMs);
+        res["ctimeMs"] = Napi::Number::New(env, ctimeMs);
+        res["mtimeMs"] = Napi::Number::New(env, mtimeMs);
+        res["birthtimeMs"] = Napi::Number::New(env, birthtimeMs);
+        res["atime"] = Napi::Date::New(env, uint64_t(round(atimeMs)));
+        res["mtime"] = Napi::Date::New(env, uint64_t(round(mtimeMs)));
+        res["ctime"] = Napi::Date::New(env, uint64_t(round(ctimeMs)));
+        res["birthtime"] = Napi::Date::New(env, uint64_t(round(birthtimeMs)));
+
+        _deferred.Resolve(res);
+    }
+};
+
+/**
+ * Statfs is an fs op
+ */
+struct Statfs : public FSWorker
+{
+    std::string _path;
+    struct statfs _statfs_res;
+
+    Statfs(const Napi::CallbackInfo& info)
+        : FSWorker(info)
+    {
+        _path = info[1].As<Napi::String>();
+        Begin(XSTR() << "Statfs " << DVAL(_path));
+    }
+    virtual void Work()
+    {
+        int r = statfs(_path.c_str(), &_statfs_res);
+        if (r) SetSyscallError();
+    }
+    virtual void OnOK()
+    {
+        DBG1("FS::Statfs::OnOK: " << DVAL(_path) << DVAL(_statfs_res.f_type) << DVAL(_statfs_res.f_bsize));
+        Napi::Env env = Env();
+        auto res = Napi::Object::New(env);
+        res["__fsword"] = Napi::Number::New(env, _statfs_res.f_type);
```

Comment:
I do not understand why you assign different values to the same properties multiple times.
2 assignments for __fsword
3 assignments for fsblkcnt
2 assignments for fsfilcnt
3 assignments for __fsword